### PR TITLE
Adding SageMakerRuntime API

### PR DESCRIPTION
**Pull request by royjit**
Finishes #166596869

*Issue #, if available:* NA

*Description of changes:* 
* Adding model and service classes to make Sagemaker runtime API
calls. Currently Sagemaker as InvokeEndpoint API.
* Changes in AWSSerialization.m to support payload "BodyBlob"
* Changes in AWSService.m to support the new endpoint from sagemaker runtime

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rohandubal** [comment]
We need to resolve the build failures. It is likely due to Swift version.

File: AWSSageMakerRuntime.podspec [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Pod::Spec.new do |s|
+  s.name         = 'AWSSageMakerRuntime'
+  s.version      = '2.2'
```
Comment (rohandubal) [inline AWSSageMakerRuntime.podspec]: We need to match this to the current SDK Version.

File: AWSSageMakerRuntime.podspec [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  s.description  = 'The AWS SDK for iOS provides a library, code samples, and documentation for developers to build connected mobile applications using AWS.'
+
+  s.homepage     = 'http://aws.amazon.com/mobile/sdk'
+  s.license      = 'Apache License, Version 2.0'
+  s.author       = { 'Amazon Web Services' => 'amazonwebservices' }
+  s.platform     = :ios, '8.0'
+  s.source       = { :git => 'https://github.com/aws/aws-sdk-ios.git',
+                     :tag => s.version}
+  s.requires_arc = true
+  s.dependency 'AWSCore', '2.2'
```
Comment (rohandubal) [inline AWSSageMakerRuntime.podspec]: Here as well.

File: AWSSageMakerRuntime.podspec (hunk 1 above)
Comment (royjit) [reply AWSSageMakerRuntime.podspec]: Fixed

File: AWSSageMakerRuntime.podspec (hunk 2 above)
Comment (royjit) [reply AWSSageMakerRuntime.podspec]: Fixed

**Comment by royjit** [comment]
> We will be having a separate PR for the CI build/ test/ release logic right?

Yes, will do a separate PR for the circleci changes

File: AWSSageMakerRuntimeTests/AWSSageMakerRuntimeTests.swift [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    
+    let WRONG_SAGEMAKER_ENDPOINT = "wrong-endpoint"
+    let INPUT_FOR_7 = testImageBufferForValue_7().data(using: .utf8)
+    let INCORRECT_INPUT = "123".data(using: .utf8)
+
+    var sageMakerEndpoint: String?
+    
+    override func setUp() {
+        super.setUp()
+        AWSTestUtility.setupCrdentialsViaFile()
```
Comment (palpatim) [inline AWSSageMakerRuntimeTests/AWSSageMakerRuntimeTests.swift]: nit: s/Crdentials/Credentials/g

File: AWSSageMakerRuntimeTests/AWSSageMakerTests.swift [hunk 4]
```diff
@@ -0,0 +1,105 @@
+//
```
Comment (palpatim) [inline AWSSageMakerRuntimeTests/AWSSageMakerTests.swift]: What is the difference between these tests and the ones above in `AWSSageMakerRuntimeTests.swift` ?

File: AWSSageMakerRuntimeUnitTests/AWSGeneralSageMakerRuntimeTests.m [hunk 5]
```diff
@@ -0,0 +1,108 @@
+//
```
Comment (palpatim) [inline AWSSageMakerRuntimeUnitTests/AWSGeneralSageMakerRuntimeTests.m]: I assume these unit tests were auto-generated by codegen?

Comment (royjit) [reply AWSSageMakerRuntimeUnitTests/AWSGeneralSageMakerRuntimeTests.m]: Yes

File: AWSSageMakerRuntimeTests/AWSSageMakerTests.swift (hunk 4 above)
Comment (royjit) [reply AWSSageMakerRuntimeTests/AWSSageMakerTests.swift]: One is for unit test and other for integration

File: AWSSageMakerRuntimeTests/AWSSageMakerRuntimeTests.swift (hunk 3 above)
Comment (royjit) [reply AWSSageMakerRuntimeTests/AWSSageMakerRuntimeTests.swift]: Fixed

File: AWSSageMakerRuntimeTests/AWSSageMakerTests.swift (hunk 4 above)
Comment (palpatim) [reply AWSSageMakerRuntimeTests/AWSSageMakerTests.swift]: Can we move the unit tests to the "Unit Tests" directory then?

Comment (royjit) [reply AWSSageMakerRuntimeTests/AWSSageMakerTests.swift]: Sorry my bad, both where same file. I removed one now.

**Pull request merged**